### PR TITLE
Add pyopenssl recipe

### PR DESCRIPTION
**Pull request by goanpeca**

### CODE REVIEW

**Comment by goanpeca** [comment]
Depends on https://github.com/conda-forge/staged-recipes/pull/1333

**Comment by jakirkham** [comment]
Best I get busy then. 😉 Will look at it again tomorrow to see where we stand. Need some sleep first. 😴

**Comment by goanpeca** [comment]
Sure, pushing **ndg-httpsclient** and I am off

File: recipes/pyopenssl/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+test:
+  imports:
+    - OpenSSL
+    - OpenSSL.crypto
+    - OpenSSL.rand
+    - OpenSSL.SSL
+
+about:
+  home: https://github.com/pyca/pyopenssl
+  license: Apache License, Version 2.0
```
Comment (jakirkham) [inline recipes/pyopenssl/meta.yaml]: `Apache 2.0` is fine.

File: recipes/pyopenssl/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    -SSL.Connection objects, wrapping the methods of Python's portable sockets
+    -Callbacks written in Python
+    -Extensive error-handling mechanism, mirroring OpenSSL's error codes
+    and much more.
+  doc_url: https://pyopenssl.readthedocs.org/en/stable/
+  dev_url: https://github.com/pyca/pyopenssl
+
+extra:
+  recipe-maintainers:
+    - goanpeca
```
Comment (jakirkham) [inline recipes/pyopenssl/meta.yaml]: nit: Please add a terminal newline.

File: recipes/pyopenssl/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+test:
+  imports:
+    - OpenSSL
+    - OpenSSL.crypto
+    - OpenSSL.rand
+    - OpenSSL.SSL
+
+about:
+  home: https://github.com/pyca/pyopenssl
+  license: Apache 2.0
```
Comment (jakirkham) [inline recipes/pyopenssl/meta.yaml]: Could add `license_family: Apache`.

**Comment by jakirkham** [comment]
Some minor things, but will save them for the feedstock. Thanks @goanpeca .

**Pull request merged**